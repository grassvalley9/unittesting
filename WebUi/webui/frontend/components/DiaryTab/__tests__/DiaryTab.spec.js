import React from 'react'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { createSerializer } from 'enzyme-to-json'

import Diary from '../index'

import * as c from '../../../store/actionTypes'

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }))

configure({ adapter: new Adapter() })

global.mount = mount
global.shallow = shallow

describe('<DiaryTabView />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <Diary
        loadItems={() => console.log('items')}
        updateTime={() => console.log('updateTime')}
      />
    )
  })
  it(`render <Diary /> without data`, () => {
    expect(wrapper).toMatchSnapshot()
  })
})