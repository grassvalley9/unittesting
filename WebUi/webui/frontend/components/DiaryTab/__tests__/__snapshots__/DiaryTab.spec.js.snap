 Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiaryTabView /> render <Diary /> without data 1`] = `
<div>
  <Diary
    loadItems={[Function]}
    updateTime={[Function]}
  />
</div>
`;